src/api/commands/profile/ProfileRemoveCommand.ts

```diff
@@ -48,6 +51,31 @@ export class ProfileRemoveCommand extends BaseCommand implements RpcCommandInter
         return this.profileService.destroy(profileId);
     }
 
+    public async validate(data: RpcRequest): Promise<RpcRequest> {
+        if (data.params.length < 1) {
+            throw new MissingParamException('profileId');
+        }
+
+        const profileId = data.params[0];
+        if (typeof profileId !== 'number' && typeof profileId !== 'string') {
+            throw new InvalidParamException(profileId, 'number|string');
+        }
+
+        if (typeof profileId === 'number') {
+            const profile = this.profileService.findOne(profileId);
+            if (!profile) {
+                throw new NotFoundException(profileId);
```

Comment:
no need to throw, all findOne[ByX] methods throw NotFoundException if they can't find something